all-contacts/1-pdb-generation/fah-pdbs-make.pl

```diff
@@ -51,20 +59,20 @@ sub generate_pdbs_from_logfile {
         if ($run != $previous_run or $clone != $previous_clone) {
 
             # print informative statistics and reset PDB count
-            print $OUT "PROJ$Project/RUN$run/CLONE/$clone\t$current_pdbs_count PDBs created\n";
+            print $OUT "[INFO]  PROJ$Project/RUN$run/CLONE$clone\t$current_pdbs_count PDBs created\n";
             $current_pdbs_count = 0;
 
             # change to new working directory and remove all existing PDBs
-            my $workdir = "$homedir/PROJ$Project/RUN$run/CLONE$clone/";
+            my $workdir = "$homedir/$Project_Dir/RUN$run/CLONE$clone/";
             chdir $workdir;
```

Comment:
I see you do this throughout the whole script, any reason you chose this over using absolute paths?

Reply:
By "this" do you mean line 67? I'm not sure I understand your question.

Reply:
Yes. In particular your use of chdir.

Reply:
We need to `chdir` so that gmx_tools (in this case `trjconv`) can run correctly.

Reply:
`trjconv` takes in user arguments which can be relative or absolute paths.

Ex:    
    Relative: `trjconv -f file.xtc -o file-out.pdb ... etc`
    Absolute: `trjconv -f /data/fah-data/file.xtc -o /data/fah-data/pdbout/file-out.pdb`

In both cases trjconv would run exactly the same. The difference, however, is that the "relative path" version would fail if run in an incorrect directory.

You are ensuring that trjconv won't fail by calling `chdir` but this mean you are making assumptions on where (in the file system) the program is at during runtime. This can make it hard to debug issues in the case of a failure.

I'm ok with this implementation but think it is pertinent to bring this up.

Note that utilizing absolute paths would require you to restructure the script entirely...

Reply:
I agree. The subroutine `generate_pdbs_from_logfile()` has the logic preserve from the "good ol' day" so I'm reluctant to change. Since it's working as is, can we do come back to this at a later time?

Reply:
I also believe the F@H directory structure hasn't been changed. So the original assumption that the PROJ directory contains multiple RUN directories, each of which contains multiple CLONE directories is still correct.

Reply:
I also realize that using absolute paths could induce some overhead in refactoring the logging logic, including ensuring that log messages don't contain too long of a path.

Reply:
Ah, I didn't know that this implementation was already in place, then it's fine. Ideally we want to use absolute paths but it can wait. & I forgot to mention, the intuition behind my comments comes from a bad experience debugging Yessica's script that utilized similar logic steps.

Reply:
& logging "too long of a path" might be a subjective issue? From talking with Sorin about logs he seems to prefer as detailed of a log file as we can offer. Regardless, from the looks of the code I think it should work just fine.